cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/EntityImpersonator.java

```diff
@@ -39,15 +40,14 @@ public NamespacedImpersonator(NamespaceId namespaceId, Impersonator impersonator
    * @param callable callable
    * @param <T> callable return type
    * @return result of callable
-   * @throws Exception
    */
   public <T> T impersonate(final Callable<T> callable) throws Exception {
-    // todo namespaceId shouldn't be null, it's passed null only from PluginService. which needs to be updated.
-    if (namespaceId == null || namespaceId.equals(NamespaceId.SYSTEM)) {
+    // todo entityId shouldn't be null, it's passed null only from PluginService. which needs to be updated.
```

Comment:
This has been around for a few months, Shankar might already have a Jira, will confirm.